lifting-state-up/lifting-state-up.md

```diff
@@ -0,0 +1,221 @@
+        }
+
+        // Invoke the function passed in through the prop named "onUpate"
+        // This function is referenced by the name "updateStudentData" in App
+        props.onUpdate(updatedStudent);
+    }
+
+    // ... other rendering logic
+}
+```
+<!-- prettier-ignore-end -->
+
+| <div style="min-width:200px;"> Piece of Code </div> | Notes                                                                                                                                                                                                                                                          |
+| --------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `const onAttendanceButtonClick = () => { ... }`     | We're creating an event-handling function named `onAttendanceButtonClick`.                                                                                                                                                                                     |
+| `const updatedStudent = { ... }`                    | We create a variable, `updatedStudent`, to hold our updated student data.                                                                                                                                                                                      |
+| `id: props.id, ... emailData: props.email,`         | The key-value pairs in this object should match exactly what is required in `App`'s `updateStudentData` function. In that function, we're looking for the keys `id`, `nameData`, `emailData`, and `isPresentData`, to match the existing `studentData` format. |
+| `isPresentData: !props.isPresent`                   | When we click on the attendance button, the value of `isPresentData` should be the negation of `props.isPresent`.                                                                                                                                              |
+| `props.onUpdate(...);`                              | The value of `props.onUpdate` is a function. We invoke this function using parentheses `()`.                                                                                                                                                                   |
+| `updatedStudent`                                    | The function `props.onUpdate` ultimately accepts one argument, an updated student object. We can pass in our new `updatedStudent` object here.                                                                                                                 |
+
+## Attaching `onAttendanceButtonClick` to the Attendance Button
+
+Now that we have an event handler, let's ensure that we're _using_ this event handler!
```

Comment:
say defined as the foil to using
```suggestion
Now that we have defined an event handler, let's ensure that we're _using_ this event handler!
```

Reply:
Literary genius. (this is not sarcastic 😆 )